narwhals/_duckdb/dataframe.py

```diff
@@ -242,18 +242,15 @@ def _with_native(self: Self, df: duckdb.DuckDBPyRelation) -> Self:
             df, backend_version=self._backend_version, version=self._version
         )
 
-    def group_by(self: Self, *keys: str, drop_null_keys: bool) -> DuckDBGroupBy:
+    def group_by(self: Self, *keys: DuckDBExpr, drop_null_keys: bool) -> DuckDBGroupBy:
         from narwhals._duckdb.group_by import DuckDBGroupBy
 
         return DuckDBGroupBy(self, keys, drop_null_keys=drop_null_keys)
 
     def rename(self: Self, mapping: Mapping[str, str]) -> Self:
         df = self.native
-        selection = [
-            f"{name} as {mapping[name]}" if name in mapping else name
-            for name in df.columns
-        ]
-        return self._with_native(df.select(", ".join(selection)))
+        selection = (col(name).alias(mapping.get(name, name)) for name in df.columns)
+        return self._with_native(self.native.select(*selection))
```

Comment:
The issue with existing implementation is for dashed names. If mapping is `{"foo": "foo-bar"}`, then the selection evaluates to `"foo as foo-bar"` and ends up failing!

Reply:
ooh nice! daily reminder to use the python api more and handwritten sql less 😄

Reply:
can we make a separate PR to fix the issue with the current implementation, so we can merge that immediately?

Reply:
Sure thing!

Reply:
There you go: #2382